Libs/MRML/Core/vtkMRMLSliceCompositeNode.h

```diff
@@ -67,31 +68,55 @@ class VTK_MRML_EXPORT vtkMRMLSliceCompositeNode : public vtkMRMLNode
   void SetLabelVolumeID(const char* id);
   void SetReferenceLabelVolumeID(const char *id) { this->SetLabelVolumeID(id); }
 
+  enum
+  {
+    LayerNone = -1,
+    LayerBackground = 0,
+    LayerForeground = 1,
+    LayerLabel = 2,
+    Layer_Last // must be last
+  };
+
+  vtkMRMLVolumeNode* GetLayerVolume(unsigned int layerIndex);
+  void SetLayerVolume(unsigned int layerIndex, vtkMRMLVolumeNode* volumeNode);
+
+  const char* GetLayerVolumeID(unsigned int layerIndex);
+  void SetLayerVolumeID(unsigned int layerIndex, const char* volumeNodeID);
+
   ///
   /// Compositing mode for foreground and background can be alpha
   /// blending, reverse alpha blending, addition, or subtraction
-  vtkGetMacro (Compositing, int);
-  vtkSetMacro (Compositing, int);
+  int GetCompositing();
+  void SetCompositing(int value);
 
   ///
   /// Configures the behavior for blending layers.
   /// The default value is true, indicating that the layers will be clipped to fit the background.
   vtkGetMacro (ClipToBackgroundVolume, bool);
   vtkSetMacro (ClipToBackgroundVolume, bool);
 
-  ///
+  /// @{
+  /// Opacity of layer N over layer N-1
+  /// \note Only Foreground and Label opacity are saved into the MRML Scene
+  double GetLayerOpacity(unsigned int layerIndex);
+  void SetLayerOpacity(unsigned int layerIndex, double value);
```

Comment:
Should this API be `SetNthLayerOpacity` `GetNthLayerOpacity` for consistency with other properties like that?

Reply:
Considering that:
* layer is implicitly associated with the concept of ordering
* the method `vtkMRMLSliceLogic::GetLayerVolumeNode` already exist (and is used in project like [this one](https://github.com/petrobras/GeoSlicer/blob/54e6deb9cd938ac7b6cb55ac66a7cbc0e1c800a3/tools/deploy/Customizer.py#L1494))

I am not sure what is best.

The time being, I am deprecating `vtkMRMLSliceLogic::GetLayerVolumeNode` and using `GetNth/SetNth` terminology.

We will settle this during review after I updated this pull request. :rocket:

Reply:
I guess that for simple use cases when modules are only manipulating a background volume, some helper methods could make sense.

Something like `GetFirst` /  `SetFirst`

Reply:
Since we already have `vtkMRMLSliceLogic::GetFirstVolumeNode`, I am marking this as resolved.